lib/Net/OpenSSH/Compat/SSH2.pm

```diff
@@ -466,14 +466,14 @@ sub _slave_exited {
 sub DESTROY {
     my $chan = shift;
     my $ch = $chan->_hash;
-    $chan->close if $ch->{state} eq 'exec';
+    $chan->close if $ch->{state} eq 'exec' and defined $chan;
     $chan->SUPER::DESTROY;
 }
 
 sub wait_closed {
     my $chan = shift;
     my $ch = $chan->_hash;
-    shift->close if $ch->{state} eq 'exec';
+    shift->close if $ch->{state} eq 'exec' and defined shift;
```

Comment:
i got message which says: "Can't call method "close" on an undefined value at /usr/local/lib/site_perl/5.24.0/Net/OpenSSH/Compat/SSH2.pm line 476." in this lines.